### DISCUSSION TITLE
Dúvida sobre Juros, Multa e Desconto no QR Com Vencimento

### DISCUSSION THREAD

**Discussion by mzavarize** [bacen/pix-api#383] [Q&A]
Boa tarde,

Estou com uma dúvida, no caso de eu ser um PSP Recebedor de QR Code com Vencimento, o PSP pagador irá fazer a consulta através da minha location passando o DPP (caso não informe usarei a data atual) e codMunicipio (se não vier, devo considerar somente feriados nacionais), calcularemos os valores e devolveremos no JWS, mas quando vier a efetivação do pagamento pelo PACS008, virá só o valor total, não receberemos o que está compondo aquele valor e para eu calcular, não há o código do município, assim não consigo certamente verificar qual é o valor para juros, multa e desconto, para demonstrar ao recebedor como foi composto aquele valor, poderá ainda haver o problema de haver a consulta com uma dpp e o mesmo ser efetivado em outra data (não deveria ocorrer, mas não há nenhum impedimento). Como estão tratando essa questão?

**Comment by rubenskuhl** [comment]
Como o @ninrod já explicou em outros tópicos, como o codMun não transita na PACS008, deve-se guardar o codMun dos acessos ao payload. O que eu não lembro é se no acaso de acessos com codMun diferentes se vale o primeiro ou o último. 

Já da data se na data do efetivo pagamento a quantia transferida corresponde à especificada na cobrança, o pagamento deve ser aceito e caso não, recusado. Independente de que DPPs tenham sido informadas no passado.

**Reply by mzavarize** [reply]
Sim, isso é uma alternativa, mas por exemplo o pagador faz uma consulta em dois aplicativos diferente que ele possui que por algum motivo vem códigos de município distintos ou até consulta de contas diferentes, vamos ter que calcular com os códigos que tem e verificar se um deles estaria correto, mas isso demanda tempo de processamento na validação, e temos os SLAs a ser cumpridos.

**Comment by mzavarize** [comment]
Olá @rubenskuhl, me desculpe, cometi um equivoco na informação acima, relendo vi que postou "O que eu não lembro é se no acaso de acessos com codMun diferentes se vale o primeiro ou o último.". Muito obrigado.
Uma outra alternativa, já que vou gravar o código do município, poderia gravar os dados calculados nas consultas e validar o valor recebido contra esses cálculos, pois entendo que se a data de pagamento for alterada, deverá fazer uma nova consulta para receber os valores, assim diminui a chance de rejeição. Não sei se teria alguma não conformidade desta maneira. Mais uma vez, obrigado.

**Reply by rubenskuhl** [reply]
> Olá @rubenskuhl, me desculpe, cometi um equivoco na informação acima, relendo vi que postou "O que eu não lembro é se no acaso de acessos com codMun diferentes se vale o primeiro ou o último.". Muito obrigado.
> Uma outra alternativa, já que vou gravar o código do município, poderia gravar os dados calculados nas consultas e validar o valor recebido contra esses cálculos, pois entendo que se a data de pagamento for alterada, deverá fazer uma nova consulta para receber os valores, assim diminui a chance de rejeição. Não sei se teria alguma não conformidade desta maneira. Mais uma vez, obrigado.

Mas a opção que você citou(mesmo que manifestando não gostar dela) de guardar todos os codMun também seria válida... vamos ver o que diz o @ninrod. 

Acho que guardando o codMun, que é necessário, gravar os demais ou calcular todos na hora é uma decisão de normalização de dados(https://pt.wikipedia.org/wiki/Normaliza%C3%A7%C3%A3o_de_dados) que não afeta interoperabilidade. Eu pessoalmente guardaria só o codMun, mas cada um, cada um...

**Comment by rubenskuhl** [comment]
Em tempo: guardar só o primeiro codMun seria um problema, pois pode impedir um usuário pagador de corrigir uma informação errada. Então as opções que fazem mais sentido são mesmo o último codMun usado ou todos.

**Reply by ninrod** [reply]
Pessoal, boa tarde. 

[Essa](https://github.com/bacen/pix-api/issues/199) é a thread que vocês querem ler.

**Reply by mzavarize** [reply]
Olá Obrigado pela resposta de todos, pelo que vi na thread postada pelo @ninrod , ele sugeriu armazenar os valores também, mas como @rubenskuhl é mais uma decisão interna na normalização de dados. Mais uma vez, obrigado a todos.!
